src/Package_Command.php

```diff
@@ -1103,14 +1137,18 @@ private function guess_version_constraint_from_tag( $tag ) {
 	 * Avoids authorization failures when accessing various sites.
 	 */
 	private function set_composer_auth_env_var() {
-		$changed = false;
+		$changed       = false;
 		$composer_auth = getenv( 'COMPOSER_AUTH' );
-		if ( ! $composer_auth || ! ( $composer_auth = json_decode( $composer_auth, true /*assoc*/ ) ) || ! is_array( $composer_auth ) ) {
-			$composer_auth = array();
+		if ( ! $composer_auth ) {
```

Comment:
@jrfnl I have addressed rest of the changes, I am not getting this, since it had an assignment, and `$composer_auth` already declared above, I couldn't add it above so updated it this way, should I add `false !==` check for these?

Reply:
@thrijith Yes, that was my line of thinking `if ( false !== $composer_auth ) {` as `getenv()` will return `false` if the environment variable doesn't exist.

Though if I look at what happens after that with the variable, something like: `if ( is_string( $composer_auth ) && '' !== $composer_auth ) {` might be even better.

Basically what I'm suggesting is: check for what you actually want to know a variable complies with.